core/src/commonMain/kotlin/com/powersync/db/internal/InternalDatabaseImpl.kt

```diff
@@ -189,21 +160,31 @@ internal class InternalDatabaseImpl(
     override fun <RowType : Any> watch(
         sql: String,
         parameters: List<Any?>?,
+        throttleMs: Long?,
         mapper: (SqlCursor) -> RowType,
-    ): Flow<List<RowType>> {
-        val tables =
-            getSourceTables(sql, parameters)
-                .map { toFriendlyTableName(it) }
-                .filter { it.isNotBlank() }
-                .toSet()
-        return watchQuery(
-            query = sql,
-            parameters = parameters?.size ?: 0,
-            binders = getBindersFromParams(parameters),
-            mapper = mapper,
-            tables = tables,
-        ).asFlow().mapToList(scope.coroutineContext)
-    }
+    ): Flow<List<RowType>> =
+        // Use a channel flow here since we throttle (buffer used under the hood)
+        // This causes some emissions to be from different scopes.
+        channelFlow {
+            // Fetch the tables asynchronously with getAll
+            val tables =
+                getSourceTables(sql, parameters)
+                    .filter { it.isNotBlank() }
```

Comment:
Do we still need this given that the `ps_data__` stripping is gone?

Reply:
I'm not entirely sure why this method returns potentially blank table names, but I think it still makes sense to filter blank strings from this list.

Reply:
Yeah I don't think `sqlite_master.name` can be empty, but it also doesn't hurt so we can leave it is.